lib/index.js

```diff
@@ -31,10 +32,24 @@ export const httpClientHandler = {
       throw new Error('NotFoundError');
     }
 
-    return result.data;
+    return _getJsonData(result);
   }
 };
 
+async function _getJsonData(result) {
+  if(result.data) {
+    return result.data;
+  }
+
+  try {
+    return await result.json();
+  } catch(e) {
+    const {url} = result;
+    logger.error(`DataError: could not parse JSON from: "${url}"`);
```

Comment:
Is this something that should be logged?  Would it be good enough to let calling code handle the error?  Maybe add the URL to it.